dozer/cogs/_utils.py

```diff
@@ -289,4 +301,56 @@ def decorator(func):
             func.__required_permissions__ = discord.Permissions()
             func.__required_permissions__.update(**required)
         return func
+
     return decorator
+
+
+class PrefixHandler:
+    """Handles dynamic prefixes"""
+
+    def __init__(self, default_prefix):
+        self.default_prefix = default_prefix
+        self.prefix_cache = {}
+
+    def handler(self, bot, message):
+        """Process the dynamic prefix for each message"""
+        dynamic = self.prefix_cache.get(message.guild.id)
+        return dynamic if dynamic else self.default_prefix
+
+    async def refresh(self):
+        """Refreshes the prefix cache"""
+        prefixes = await DynamicPrefixEntry.get_by()  # no filters, get all
+        for prefix in prefixes:
+            self.prefix_cache[prefix.guild_id] = prefix.prefix
```

Comment:
If a dynamic prefix is deleted, it needs to be removed from the cache. `self.prefix_cache.clear()` before this loop would do the trick.

Reply:
A dynamic prefix will never get deleted, to go back to default it'll just get set to '&' but it will still be a "dynamic prefix" just its a default dynamic prefix